src/openrct2/actions/MazeSetTrackAction.hpp

```diff
@@ -99,7 +99,7 @@ struct MazeSetTrackAction : public GameActionBase<GAME_COMMAND_SET_MAZE_TRACK, G
             return res;
         }
 
-        if ((_z & 0xF) != 0)
+        if ((_z & 0xF) != 0 && _mode == GC_SET_MAZE_TRACK_BUILD)
```

Comment:
If we're demolishing a maze, what is `_mode` set to?

Reply:
It is set on GC_SET_MAZE_TRACK_FILL. 
Somehow, on a single block maze with one "track" unfilled (the default when placing the first block), it calls it 3 times.

With that said, it will probably seem like a weird check to make if anyone ever reads that code in the future. Should this part be rethought or would a simple comment do the trick for now?

P.S.: I also found it was possible to completely fill the first block, which makes it disappear, but to still keep the maze ride and demolish it, resulting in a negative money gain. Bug is present in develop. I'll raise an issue with all the details later.

Reply:
Well, if there is no proper way of doing it, please add a comment that clearly explains it. That way, if someone comes along with a better way, they will know why you did it, and what the possible side effects of changing it can be.

The error you describe in your P.S., is that present in this PR, or has it been solved by this PR as well?

Reply:
It was fixed in PR #8247